src/App.tsx

```diff
@@ -1,41 +1,63 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getAllUsers } from './api/api';
+import { User } from './react-app-env';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
+const App: React.FC = () => {
+  const [allUsers, setAllUsers] = useState<User[]>([]);
+  const [currentUser, setCurrentUser] = useState('0');
+  const [selectedPostId, setSelectedPostId] = useState<number>();
 
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
+  useEffect(() => {
+    getAllUsers()
+      .then(response => setAllUsers(response));
+  }, []);
```

Comment:
Don't forget to use `.catch` method. This helps in making the code more robust by addressing potential errors and providing a way to handle them gracefully.
```suggestion
  useEffect(() => {
    getAllUsers()
      .then(response => setAllUsers(response))
      .catch(error =>console.error(error));
  }, []);
```

Also you can add some loading moments to combine additional `.then` with `.finally`. [Doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally)